Bug report: `getIndirectPullRequests()` gets `TypeError: Cannot read property 'commitId' of undefined`

**Describe the bug**

This bug is related to `XplatGenerateReleaseNotes:v3.*` (exact: `3.35.7`)

When enabling the `Advanced > Get Indirect Pull Requests`, the following error raised.

I think this happens because the [`var foundPR = allPullRequests.find( e => e.lastMergeCommit.commitId === cs.commitId);`](https://github.com/rfennell/AzurePipelines/blob/ad5d23c33e0d6e462be867f2da823c346bb817ee/Extensions/XplatGenerateReleaseNotes/V3/ReleaseNotesFunctions.ts#L1296) doesn't check the existence of `e.lastMergeCommit`, whereas the other place [`if (pr.lastMergeCommit) { if (pr.lastMergeCommit.commitId === commit.id) {`](https://github.com/rfennell/AzurePipelines/blob/ad5d23c33e0d6e462be867f2da823c346bb817ee/Extensions/XplatGenerateReleaseNotes/V3/ReleaseNotesFunctions.ts#L1258-L1265) does check.

I'm not sure why this happenes but I couldn't find this PR (`1038`) in all of our PRs and repos.
Therefore, I guess this PR becomes unavailable becasuse its associated repo was deleted.

**To Reproduce**

Not sure how

**Expected behavior**

Should bypass the `commitId` access if the `e.lastMergeCommit` does not exists

**Screenshots**

N/A

**Logs**
```
2020-12-17T06:30:16.4837493Z Found 183 Azure DevOps PRs in the repo
2020-12-17T06:30:16.4842879Z Checking for PRs associated with the commit 215d95a9b2f88ab6d0a8b5b7ada1f7b526ed940e
2020-12-17T06:30:16.4843475Z - PR 1170 matches the commit 215d95a9b2f88ab6d0a8b5b7ada1f7b526ed940e
2020-12-17T06:30:16.4843929Z - PR 1038 does not have a lastMergeCommit
2020-12-17T06:30:16.4844213Z Checking for PRs associated with the commit 0968c4bebedb9affc78a42ba5d39ff9fd8e69184
2020-12-17T06:30:16.4844639Z - PR 1038 does not have a lastMergeCommit
2020-12-17T06:30:16.4845073Z Enriching known Pull Requests
2020-12-17T06:30:16.7970936Z ##[debug]Checking for tags for 1170
2020-12-17T06:30:16.9202928Z ##[debug]Added 0 work items for 1170
2020-12-17T06:30:17.0463609Z ##[debug]Added 1 commits for 1170, note this includes commits on the PR source branch not associated directly with the build
2020-12-17T06:30:17.0464600Z Checking the CS associated with the PRs to see if they are inturn associated PRs
2020-12-17T06:30:17.0496763Z ##[error]TypeError: Cannot read property 'commitId' of undefined
2020-12-17T06:30:17.0506378Z ##[debug]Processed: ##vso[task.issue type=error;]TypeError: Cannot read property 'commitId' of undefined
```

**What is the issue see:**
 - `XplatGenerateReleaseNotes:3.35.7`

--- a/Extensions/XplatGenerateReleaseNotes/V3/ReleaseNotesFunctions.ts
+++ b/Extensions/XplatGenerateReleaseNotes/V3/ReleaseNotesFunctions.ts
@@ -1293,7 +1293,7 @@ export async function generateReleaseNotes(
                         const pr = globalPullRequests[prIndex];
                         for (let csIndex = 0; csIndex < pr.associatedCommits.length; csIndex++) {
                             const cs =  pr.associatedCommits[csIndex];
-                            var foundPR = allPullRequests.find( e => e.lastMergeCommit.commitId === cs.commitId);
+                            var foundPR = allPullRequests.find( e => e.lastMergeCommit && e.lastMergeCommit.commitId === cs.commitId);
                             if (foundPR) {
                             agentApi.logInfo(`Found the PR ${foundPR.pullRequestId} associated wth ${cs.commitId} added to the 'inDirectlyAssociatedPullRequests' array`);
                             inDirectlyAssociatedPullRequests.push(<EnrichedGitPullRequest>foundPR);

--- a/Extensions/XplatGenerateReleaseNotes/readme.md
+++ b/Extensions/XplatGenerateReleaseNotes/readme.md
@@ -342,7 +342,9 @@ This task makes many calls to the Azure DevOps REST API, the volume depends on t
 Error: connect ETIMEDOUT 13.107.42.18:443
 ```
 
-At present, even though the task utilizes retry logic, there is no way to recover from these timeouts. The only solution is to retry the failed job, this will usually work (eventually)
+This is an issue with the underlying Azure DevOps Node SDK or REST API endpoints, not this task. Hence, an issue has been raised in the the appropriate [Repo #425](https://github.com/microsoft/azure-devops-node-api/issues/425)
+
+The best workaround at present is to always place this task, and any associated tasks e.g. one that upload the generated release notes to a WIKI, in a dedicated YML pipeline job. This allows the task to be easily retried without rerunning the whole pipeline.
 
 ## OAUTH Scope limiting what Associated Items can be seen
 This task uses the build agent's access OAUTH token to access the Azure DevOps API. The permissions this identity has is dependant upon the [the Job authorization scope](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/access-tokens?view=azure-devops&tabs=yaml#job-authorization-scope).

